### PR TITLE
Add external links to .links.json

### PR DESCRIPTION
**Pull request by live1206**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/152392

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5413)

### CODE REVIEW

File: src/docfx/lib/HtmlUtility.cs [hunk 1]
```diff
@@ -269,9 +269,9 @@ public static string CreateHtmlMetaTags(JObject metadata, ICollection<string> ht
             return result.ToString();
         }
 
-        public static HtmlNode AddLinkType(this HtmlNode html, string locale)
+        public static HtmlNode AddLinkType(this HtmlNode html, string locale, Action<string> addLinkAction = null)
         {
-            AddLinkType(html, "a", "href", locale);
+            AddLinkType(html, "a", "href", locale, addLinkAction);
```
Comment (OsmondJiang) [inline src/docfx/lib/HtmlUtility.cs]: will this cause duplicated links?

Comment (live1206) [reply src/docfx/lib/HtmlUtility.cs]: Yes, could do a fix for it, ignore `external` link-type during resolving link. But the inclusion of `external` links can not be fixed.

**Comment by herohua** [comment]
would it make more sense to put the logic in `LoadMarkdown`?

**Comment by yufeih** [comment]
See LinkExtension.cs

**Comment by live1206** [comment]
> would it make more sense to put the logic in `LoadMarkdown`?

This is not only for `.md` files, consider a `JSON/YAML` file with a field which is `Markdown`.

**Comment by live1206** [comment]
> This might not be the best place to process the link. Based on the test case, it seems markdown autolinks are not processed by LinkResolver, I feel like capturing autolinks and have it go though the same link resolve process is a better fix.

Consider a user written link(`<a href="link"></a>`) within HTML block, would it be parsed as autolinks during markup?

**Comment by yufeih** [comment]
html links are not markdown autolinks, but html links are also parsed and resolved using the same link resolve process, so you can write **a href=a.md** and we'll resolve a.md to the correct url using link resolver.

These logics are all handled in LinkExtension.cs

**Comment by yufeih** [comment]
https://github.com/dotnet/docfx/blob/v3/src/docfx/lib/markdown/LinkExtension.cs#L29

Autolinks are explicitly excluded, can try remove that check. I dont recall exactly why it is there in the first place, might be a perf optimization where we can skip external urls.

**Pull request merged**